map_creator/src/load_capability_map.cpp

```diff
@@ -0,0 +1,143 @@
+                                  count, NULL);
+    dimsm[0] = count[0];
+    dimsm[1] =  count[1];
+    hid_t memspace;
+    memspace = H5Screate_simple (RANK_OUT, dimsm, NULL);  
+    status = H5Dread (dataset, H5T_NATIVE_DOUBLE, memspace, dataspace,
+                      H5P_DEFAULT, data_out);
+    for(int i=0;i<count[0];i++){
+	vector<double> posAndQuat;
+	vector<double> enumRIAngle;
+        for(int j=0;j<2;j++)
+	    {
+		 enumRIAngle.push_back(data_out[i][j]);
+	    }
+	for(int j=2;j<9;j++)
+	    {
+		 posAndQuat.push_back(data_out[i][j]);
+	    }
+        for(int j=9;j<10;j++)
+	    {
+		 enumRIAngle.push_back(data_out[i][j]);
+	    }
+        
+        sphereColor.insert(pair<vector<double>, vector<double> >(posAndQuat, enumRIAngle));
+    }
+  }
+};
+
+
+int main(int argc, char **argv)
+{
+    if (argc<2){
+	ROS_ERROR_STREAM("Please provide the name of the Capability map. If you have not created it yet, Please create the map by running the create cpability map launch file in map_creator");
+	return 1;
+    } 
+    else{
+	
+    ros::init(argc, argv, "workspace");
+    ros::NodeHandle n;
+    bool latchOn = 1;
+    ros::Publisher workspace_pub = n.advertise<map_creator::capability>("capability_map", 1);
+    
+    ros::Rate loop_rate(10);
+  
+      
+      time_t startit,finish;
+      time (&startit);
+      const char* FILE = argv[1];
+      hid_t file, cap_group, cap_dataset, attr; 
+      file = H5Fopen (FILE, H5F_ACC_RDONLY, H5P_DEFAULT);
+
+//capability dataset
+
+      cap_group = H5Gopen (file, CAP_GROUPNAME, H5P_DEFAULT);
+      cap_dataset = H5Dopen (cap_group, CAP_DATASETNAME, H5P_DEFAULT);
+      multimap<vector<double>, vector<double> > sphereColor;
+      loadHD5 hd5;
+      hd5.h5ToMultiMapCap(cap_dataset, sphereColor);
+
+
+//Resolution Attribute
+      float res;
+      attr = H5Aopen(cap_dataset,"Resolution",H5P_DEFAULT);
+      herr_t ret = H5Aread(attr, H5T_NATIVE_FLOAT, &res);
+
+
+//Closing resources
+      H5Aclose(attr);
+      H5Dclose(cap_dataset);
+      H5Gclose(cap_group);
+      H5Fclose(file);
+	
+
+//Creating messages
+
+     
+
+      map_creator::capability cp;
+      cp.header.stamp = ros::Time::now();
+      cp.header.frame_id = "/base_link";
+      cp.resolution = res;
+
+      for(multimap<vector<double>, vector<double> >::iterator it = sphereColor.begin();it !=sphereColor.end(); ++it){
+      
+	map_creator::capShape cpSp;
+        cpSp.identifier = it->second[0];
+        cpSp.ri = it->second[1];
+        cpSp.angleSFE = it->second[2];
+        cpSp.pose.position.x = it->first[0];
+        cpSp.pose.position.y = it->first[1];
+	cpSp.pose.position.z = it->first[2];
+	cpSp.pose.orientation.x = it->first[3];
+	cpSp.pose.orientation.y = it->first[4];
+	cpSp.pose.orientation.z = it->first[5];
+	cpSp.pose.orientation.w = it->first[6];
+        cp.capShapes.push_back(cpSp);
+        }
+       
+       int count = 0;
+       
+    while (ros::ok())
```

Comment:
I tried running the code before and after the while loop change and it occurred to me that you may be using the loop to update the visualization.  Is that correct?  I was unaware of this and it looks like this latest change may have broken your visualization slicing.  Is there a better way to update the visualization?  Is there an Rviz event that you could subscribe to so that visualization changes are done on a callback instead of a loop?  Then the capability map would only get published on a callback and thus only on change.  If this is too much effort, just change it back to what it was before (so the visualization works), and I will merge the PR as it is.